deployment/modules/gcp/gce/tesseract/main.tf

```diff
@@ -14,39 +14,78 @@ locals {
-      "--not_after_limit=${var.not_after_limit}",
-      "--trace_fraction=${var.trace_fraction}",
-      "--batch_max_size=${var.batch_max_size}",
-      "--batch_max_age=${var.batch_max_age}",
-      "--enable_publication_awaiter=${var.enable_publication_awaiter}",
-      "--accept_sha1_signing_algorithms=true",
-      "--rate_limit_old_not_before=${var.rate_limit_old_not_before}",
-      "--rate_limit_dedup=${var.rate_limit_dedup}"
-    ]
-    tty : true # maybe remove this
-  }
+data "google_compute_image" "cos" {
+  family  = "cos-beta"
+  project = "cos-cloud"
+}
 
-  restart_policy = "Always"
+locals {
+  tesseract_args = join(" ", [
+         "-logtostderr",
+         "-v=2",
+         "-http_endpoint=:80",
+         "-bucket=${var.bucket}",
+         "-spanner_db_path=${local.spanner_log_db_path}",
+         "-spanner_antispam_db_path=${local.spanner_antispam_db_path}",
+         "-roots_pem_file=/bin/test_root_ca_cert.pem",
+         "-origin=${var.base_name}${var.origin_suffix}",
+         "-signer_public_key_secret_name=${var.signer_public_key_secret_name}",
+         "-signer_private_key_secret_name=${var.signer_private_key_secret_name}",
+         "-inmemory_antispam_cache_size=256k",
+         "-not_after_start=${var.not_after_start}",
+         "-not_after_limit=${var.not_after_limit}",
+         "-trace_fraction=${var.trace_fraction}",
+         "-batch_max_size=${var.batch_max_size}",
+         "-batch_max_age=${var.batch_max_age}",
+         "-enable_publication_awaiter=${var.enable_publication_awaiter}",
+         "-accept_sha1_signing_algorithms=true",
+         "-rate_limit_old_not_before=${var.rate_limit_old_not_before}",
+         "-rate_limit_dedup=${var.rate_limit_dedup}",
+  ])
+
+  container_name = "tesseract-${var.base_name}"
+
+  cloud_init = <<EOT
+    #cloud-config
+
+    users:
+      - name: tesseract
+        uid: 2000
+        groups: docker # Add the user to the Docker group
+
+    write_files:
+      - path: /etc/systemd/system/config-firewall.service
+        permissions: 0644
+        owner: root
+        content: |
+          [Unit]
+          Description=Configures the host firewall
+          
+          [Service]
+          Type=oneshot
+          RemainAfterExit=true
+          ExecStart=/sbin/iptables -A INPUT -p tcp --dport 80 -j ACCEPT
+      - path: /etc/systemd/system/tesseract.service
+        permissions: 0644
+        owner: root
+        content: |
+          [Unit]
+          Description=Run TesseraCT
+          Wants=gcr-online.target docker.socket config-firewall.service
+          After=gcr-online.target docker.socket config-firewall.service
+
+          [Service]
+          ExecStartPre=sudo -u tesseract /usr/bin/docker-credential-gcr configure-docker --registries ${var.location}-docker.pkg.dev
+          ExecStart=sudo -u tesseract -E /usr/bin/docker run --rm -u 2000 --name=${local.container_name} -p 80:80 --log-driver=gcplogs ${var.server_docker_image} ${local.tesseract_args}
```

Comment:
Maybe explain why `--log-driver=gcplogs` is required? Is it not enabled by default?